pkgs/development/compilers/dotnet/sdk/vmr.nix

```diff
@@ -0,0 +1,189 @@
+  targetArch = lib.elemAt (lib.splitString "-" targetRid) 1;
+
+in clangStdenv.mkDerivation rec {
+  pname = "dotnet-sdk-vmr";
+  version = manifestJson.release;
+
+  src = fetchFromGitHub {
+    owner = "dotnet";
+    repo = "dotnet";
+    rev = "refs/tags/v${version}";
+    inherit hash;
+  };
+
+  postUnpack = ''
+    # Sets all timestamps to Jan 1 1980, the earliest mtime zips support.
+    find $sourceRoot -exec touch -t 198001010000.00 {} +
+  '';
+
+  sbArtifacts = dotnetSdk.artifacts;
+
+  nativeBuildInputs = [
+    jq
+    yq
+    curl
+    git
+    cmake
+    pkg-config
+  ];
+
+  buildInputs = [
+    llvm
+    zlib
+    icu
+  ] ++ lib.optional clangStdenv.isLinux lttng-ust_2_12;
+
+  patches = [
+    ./fix-aspnetcore-portable-linux-build.patch
+  ];
+
+  postPatch = ''
+    # set the sdk version in global.json to match the bootstrap sdk
+    jq '(.tools.dotnet=$dotnet)' global.json --arg dotnet "$(${dotnetSdk}/bin/dotnet --version)" > global.json~
+    mv global.json{~,}
+
+    # set the url used by prep.sh to point to the nix store
+    xq -x \
+      '.Project.PropertyGroup |= map(if has("PrivateSourceBuiltArtifactsUrl") then .PrivateSourceBuiltArtifactsUrl=$url else . end)' \
+      eng/Versions.props \
+      --arg url "file://$(echo "${dotnetSdk.artifacts}"/*.tar.gz)" \
+      > eng/Versions.props~
+    mv eng/Versions.props{~,}
+
+    patchShebangs $(find -name \*.sh -type f -executable)
+
+    # I'm not sure why this is required, but these files seem to use the wrong
+    # property name.
+    sed -i 's:\bVersionBase\b:VersionPrefix:g' \
+      src/xliff-tasks/eng/Versions.props
+
+    substituteInPlace \
+      src/runtime/src/native/libs/System.Net.Security.Native/pal_gssapi.c \
+      --replace '"libgssapi_krb5.so.2"' '"${libkrb5}/lib/libgssapi_krb5.so.2"'
+
+    substituteInPlace \
+      src/runtime/src/native/libs/System.Globalization.Native/pal_icushim.c \
+      --replace '"libicui18n.so"' '"${icu}/lib/libicui18n.so"' \
+      --replace '"libicuuc.so"' '"${icu}/lib/libicuuc.so"' \
+      --replace 'libicuucName[64]' 'libicuucName[256]' \
+      --replace 'libicui18nName[64]' 'libicui18nName[256]'
+  '';
+
+  prepFlags = [];
+
+  configurePhase = ''
+    runHook preConfigure
+
+    # The build process tries to overwrite some things in the sdk (e.g.
+    # SourceBuild.MSBuildSdkResolver.dll), so it needs to be mutable.
+    cp -Tr ${dotnetSdk} .dotnet
+    chmod -R +w .dotnet
+    # Sets all timestamps to Jan 1 1980, the earliest mtime zips support.
+    # find .dotnet -exec touch -t 198001010000.00 {} +
+
+    HOME=$(pwd)/fake-home \
+      ./prep.sh $prepFlags
+
+    runHook postConfigure
+  '';
+
+  dontUseCmakeConfigure = true;
+
+  # https://github.com/NixOS/nixpkgs/issues/38991
+  # bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
+  LOCALE_ARCHIVE = lib.optionalString clangStdenv.isLinux
+      "${glibcLocales}/lib/locale/locale-archive";
```

Comment:
But does that even matter in the end or is it just a warning?

Reply:
It actually fails the build due to something upstream:

```
    /build/source/src/source-build-reference-packages/artifacts/source-build/self/src/eng/SourceBuild.props(20,5): error MSB3073: The command "./build.sh --configuration Release /bl:/build/source/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/sourcebuild-dependency-projects.binlog /p:LocalNuGetPackageCacheDirectory=/build/source/artifacts/obj/source-build-reference-package-cache /p:BuildDependencyPackageProjects=true /p:SetUpSourceBuildIntermediateNupkgCache=true /p:DotNetBuildFromSource=true /p:ArcadeInnerBuildFromSource=true /p:MicrosoftNetCoreIlasmPackageRuntimeId=linux-x64" exited with code 1. [/build/source/prereqs/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/8.0.0-beta.23463.1/tools/Build.proj]
    ##vso[task.logissue type=error;sourcepath=/build/source/src/source-build-reference-packages/artifacts/source-build/self/src/eng/SourceBuild.props;linenumber=20;columnnumber=5;code=MSB3073;]The command "./build.sh --configuration Release /bl:/build/source/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/sourcebuild-dependency-projects.binlog /p:LocalNuGetPackageCacheDirectory=/build/source/artifacts/obj/source-build-reference-package-cache /p:BuildDependencyPackageProjects=true /p:SetUpSourceBuildIntermediateNupkgCache=true /p:DotNetBuildFromSource=true /p:ArcadeInnerBuildFromSource=true /p:MicrosoftNetCoreIlasmPackageRuntimeId=linux-x64" exited with code 1.

    Build FAILED.

    /nix/store/q8qq40xg2grfh9ry1d9x4g7lq4ra7n81-bash-5.2-p21/bin/sh : error : setlocale: LC_ALL: cannot change locale (en_US.UTF-8) [/build/source/prereqs/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/8.0.0-beta.23463.1/tools/Build.proj]
    /nix/store/q8qq40xg2grfh9ry1d9x4g7lq4ra7n81-bash-5.2-p21/bin/bash : error : setlocale: LC_ALL: cannot change locale (en_US.UTF-8) [/build/source/prereqs/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/8.0.0-beta.23463.1/tools/Build.proj]
```